packages/@vuepress/shared-utils/src/getPermalink.ts

```diff
@@ -12,6 +12,20 @@ interface PermalinkOption {
 function removeLeadingSlash (path: string) {
   return path.replace(/^\//, '')
 }
+
+function parseDateString (dateString: string): [number, number, number] {
+  const [
+    yearStr,
+    monthStr,
+    dayStr
+  ] = dateString.split('-')
```

Comment:
Defining date in front matter with different format will cause parser to return different type of value. For example, the type of parsed value  of `2020-02-02` is `Date` (Strictly, it's an Object), whereas `2020-2-2` will be `String`.  

The point is `split` is not in `Date`'s prototype, it cant be invoked.

Reply:
I was trying to trace the code back to where that could occur... the furthest place I could tell (from reading the code statically) was from `core/lib/node/util/index.js`. There's an `inferDate` function with the following:

```
if (frontmatter.date) {
  return frontmatter.date
} else (...)
  ...
}
```

I'm guessing `frontmatter.date` could be either a string or a Date instance?

Nonetheless, I've updated the parser to support `Date`.

Reply:
FLY, VuePress' using `gray-matter` which depends `js-yaml` which follows standard `yaml` types, so all available formats are [here](https://yaml.org/type/timestamp.html). Thus, single digit format will not be considered as `datetime`, that's why it return a string.

Reply:
I checkout your last commit. It wont get error, but it still get the wrong date in EST.

Reply:
If `YYYY-MM-DD` format was what was causing problems in the first place and that gets parsed into a `Date` instance, I suspect that that `gray-matter` is the root cause in that case.

I'll try to see if I can investigate further down that path.

In the meanwhile, are you able to confirm if it address the incorrect date issue when date is inferred from path name?

Reply:
I guess so. According to the definition:
> If the time zone is omitted, the timestamp is assumed to be specified in UTC. The time part may be omitted altogether, resulting in a date format. In such a case, the time part is assumed to be 00:00:00Z (start of day, UTC).

Reply:
And I think they're correct to be unassuming and default to UTC.

I now see two paths to approach this:
- Follow the way of the crowd and force all dates to be in UTC
- Adjust to the user's local timezone by:
  1. Forcing all dates to be in UTC
  2. Offset by `Date.getTimezoneOffset()`

I think option two would make it less confusing to users, but requires more gymnastics. What do you think?

Reply:
Whichever way the team decides, it won't hurt my feelings, but I've pushed the code for option 2 if that's the desired direction.

I couldn't find a straightforward way to unit test within the codebase, but I did a cheap test in my console and seems to have the desired effect:

```javascript
((date) => {
function toUtcTime (date) {
  let year = 1970
  let month = 0
  let day = 1

  if (typeof date === 'string') {
    const [
      yearStr,
      monthStr,
      dayStr
    ] = date.split('-')

    year = parseInt(yearStr, 10)
    month = parseInt(monthStr, 10) - 1
    day = parseInt(dayStr, 10)
  } else if (date instanceof Date) {
    year = date.getFullYear()
    month = date.getMonth()
    day = date.getDate()
  }

  return Date.UTC(year, month, day)
}

function addTzOffset(utc) {
  const utcDate = new Date(utc)

  return new Date(utcDate.getTime() + (utcDate.getTimezoneOffset() * 60000))
}

return new Date(addTzOffset(toUtcTime((date))))
})('2020-1-1')
```

Reply:
I like your approach. However, it still doesn't work fine when the param is not a string. You can use @ludanxer's [repo](https://github.com/vuejs/vuepress/pull/2181#issuecomment-590673867) to test it by defining date in 2-digits, such as `2020-01-01`.

Reply:
There obviously must be a process to follow and setup properly in my local environment that I didn't do. I didn't think the fix would involve so many attempts so I didn't invest the time to look it up. If this final go doesn't work, I'll look into doing that.